site/app/controllers/pdf/PDFController.php

```diff
@@ -18,6 +18,10 @@ public function getRealPath(string $file_path, string $id): string {
         $real_path = "";
         $file_path_parts = explode("/", $file_path);
         for ($index = 1; $index < count($file_path_parts); $index++) {
+            if ($file_path_parts[$index] === "..") {
```

Comment:
Would it potentially be more effective to just add a check for ".." at the top of this function?  E.g., instead of checking every part, just use `str_contains()` to check for ".." in the file path.

Reply:
No we can't do this. `file..file.pdf` is a valid file name that would not raise security issues.

@Eceptonsu can you change this back

Reply:
Perhaps it's outside the scope of this PR, but my thought here was that we should prohibit files with `..` in them altogether to reduce the attack surface in case someone makes a bad change somewhere down the line.  If you feel strongly that we should allow files with `..` in the file name, I'm fine with that.  Perhaps we should document this information in a comment here though.

Reply:
I mean `file..pdf.pdf` is a normal file and wouldn't want it to not be renderable because of its name. I still believe the loop `..` check is necessary as that strictly checks for `..`. Maybe `/../` could be checked for but I think doing the split on `/` and checking for `..` is already pretty efficient. And `/../` might not cover all cases.

We could just leave this conversation unresolved and it'll stay visible.